qualtran/bloqs/basic_gates/su2_rotation.py

```diff
@@ -70,6 +70,12 @@ def signature(self) -> Signature:
 
     @cached_property
     def rotation_matrix(self) -> NDArray[np.complex_]:
+        if isinstance(self.lambd, sympy.Expr):
+            raise ValueError(f'Symbolic lambda not allowed: {self.lambd}')
+        if isinstance(self.phi, sympy.Expr):
+            raise ValueError(f'Symbolic phi not allowed: {self.phi}')
+        if isinstance(self.theta, sympy.Expr):
+            raise ValueError(f'Symbolic theta not allowed: {self.theta}')
```

Comment:
```suggestion
        if self.is_symbolic():
            raise ValueError(f'cannot compute rotation matrix for symbolic gate {self}')
```

Reply:
So, unfortunately, this change, while more concise, will not satisfy mypy.  The automatic type checker can't figure out that, after self.is_symbolic() the parameters are no longer expressions.  I was playing around with trying to make is_symbolic a typing.TypeGuard, but I couldn't make it work correctly given the current code structure.

One alternative, if you feel strongly, is to change to is_symbolic, then add  `# type:ignore` later, with a comment that this is guarded by is_symbolic.

Reply:
Ah that's interesting! I think then the current code you wrote is fine. 

Btw shouldn't you also check `self.global_shift`?

Reply:
can you open an issue describing your future plans? unless you're going to do that imminently

Reply:
Added global_shift.
I have no further plans.  I have no idea how to typeguard this correctly. I'm not sure it's possible.